src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -856,6 +856,9 @@ NESTED_ENTRY JIT_MonTryEnter_InlineGetThread, _TEXT
         mov                 [rsp + 8h], rcx
         .savereg            rdx, 10h
         mov                 [rsp + 10h], rdx
+	
+        push_nonvol_reg     r12 ; rcx now at [rsp + 10h]
+		
```

Comment:
Its stored at one of two places `[rsp + 10h]` or `[rsp + MON_ENTER_STACK_SIZE_INLINEGETTHREAD + 10h]` so code would be messier and use more `ifdef`?

Also it would be a stack compare for every loop spin rather than a register compare? (not sure if that would make a huge difference)